backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/objcexport/ObjCExportCodeGenerator.kt

```diff
@@ -825,6 +837,29 @@ private fun ObjCExportCodeGenerator.generateObjCImp(
     ret(genReturnValueOnSuccess(returnType))
 }
 
+private fun ObjCExportCodeGenerator.generateExceptionTypeInfoArray(baseMethod: IrFunction): LLVMValueRef =
+        exceptionTypeInfoArrays.getOrPut(baseMethod) {
+            val throwsAnnotation = findThrowsAnnotation(baseMethod)!!
+            val types = (throwsAnnotation.getValueArgument(0) as IrVararg).elements
+                    .map { (it as IrClassReference).symbol.owner as IrClass }
+
+            generateTypeInfoArray(types.toSet())
+        }.llvm
+
+private fun ObjCExportCodeGenerator.generateTypeInfoArray(types: Set<IrClass>): ConstPointer =
+        typeInfoArrays.getOrPut(types) {
+            val typeInfos = types.map { with(codegen) { it.typeInfoPtr } } + NullPointer(codegen.kTypeInfo)
+            codegen.staticData.placeGlobalConstArray("", codegen.kTypeInfoPtr, typeInfos)
+        }
+
+private fun findThrowsAnnotation(baseMethod: IrFunction): IrConstructorCall? =
+        if (baseMethod !is IrSimpleFunction || baseMethod.overriddenSymbols.isEmpty()) {
+            baseMethod.annotations.findAnnotation(KonanFqNames.throws)
+        } else {
+            // Note: frontend ensures that all topmost overridden methods have equal @Throws annotations.
```

Comment:
Can/shall we assert for that condition?

Reply:
Technically adding such an assertion should be easy.
However, considering binary compatibility (e.g. linking with other versions of libraries), it is actually possible that such a condition isn't met.
We don't have much choice in this case: either report compile (link?) error or handle this silently.
And I would strongly prefer to avoid reporting any errors from compiler, at least because we don't have proper stable infrastructure (in IR linker) to do this reliably. WDYT?

Reply:
OK, let's keep it silently processed so far. Maybe warning?

Reply:
Reporting warning makes sense, however we still don't have proper infrastructure for this. Generally I believe we should cease reporting compiler diagnostics from backend the way we are doing it now.
I expect this particular case to be quite rare, so it should be ok to postpone adding such a warning. Especially considering that I'm not completely sure about the details of new `@Throws` design, so it can change before 1.4.